stdlib/public/TensorFlow/Ops.swift

```diff
@@ -650,6 +650,10 @@ public extension Tensor {
   /// - Precondition: The tensors must have the same dimensions, except for the
   ///   specified axis.
   /// - Precondition: The axis must be in the range `-rank..<rank`.
+  @differentiable(
```

Comment:
You should be able to drop the custom VJP from `concatenating(with:alongAxis:)` now.

Reply:
The current change will not work. `Raw` ops are still not differentiable. What I meant is to define `concatenating(with:alongAxis:)` in terms of `init(concatenating:alongAxis:)`, and then differentiation will fall out. One complication though is that array literal construction is not differentiable yet.

Reply:
> The current change will not work. `Raw` ops are still not differentiable. What I meant is to define `concatenating(with:alongAxis:)` in terms of `init(concatenating:alongAxis:)`, and then differentiation will fall out. One complication though is that array literal construction is not differentiable yet.

Yes, array literal construction causes problems.
```
/Users/danielzheng/swift-tf/swift/stdlib/public/TensorFlow/Ops.swift:666:4: error: function is not differentiable
  @differentiable(where Scalar : TensorFlowFloatingPoint)
  ~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/Users/danielzheng/swift-tf/swift/stdlib/public/TensorFlow/Ops.swift:667:8: note: when differentiating this function definition
  func concatenated(with other: Tensor, alongAxis axis: Int = 0) -> Tensor {
       ^
/Users/danielzheng/swift-tf/swift/stdlib/public/TensorFlow/Ops.swift:668:34: note: cannot differentiate through a non-differentiable argument; do you want to add '.withoutDerivative()'?
    return Tensor(concatenating: [self, other], alongAxis: axis)
                                 ^
```

I'll keep the custom VJP for `concatenating(with:alongAxis:)`.

Reply:
SGTM